paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/playground/PaymentSheetPlaygroundViewModel.kt

```diff
@@ -117,6 +137,120 @@ internal class PaymentSheetPlaygroundViewModel(
         }
     }
 
+    @OptIn(ExperimentalCustomerSheetApi::class)
+    private fun prepareCustomer(
+        playgroundSettings: PlaygroundSettings
+    ) {
+        viewModelScope.launch {
+            val snapshot = playgroundSettings.snapshot()
+
+            state.value = PlaygroundState.Customer(
+                snapshot = snapshot,
+                adapter = CustomerAdapter.create(
+                    context = getApplication(),
+                    customerEphemeralKeyProvider = ::fetchEphemeralKey,
+                    setupIntentClientSecretProvider = if (
+                        snapshot[CustomerSheetPaymentMethodModeDefinition] == PaymentMethodMode.SetupIntent
+                    ) {
+                        { createSetupIntentClientSecret(it) }
+                    } else {
+                        null
+                    }
+                )
+            )
+        }
+    }
+
+    @OptIn(ExperimentalCustomerSheetApi::class)
+    private suspend fun fetchEphemeralKey(): CustomerAdapter.Result<CustomerEphemeralKey> {
+        val playgroundSettings = playgroundSettingsFlow.value
+            ?: return CustomerAdapter.Result.failure(
+                cause = Exception("Unexpected error: Playground settings were not found!"),
+                displayMessage = null,
+            )
+
+        // Snapshot before making the network request to not rely on UI staying in sync.
+        val playgroundSettingsSnapshot = playgroundSettings.snapshot()
+
+        playgroundSettingsSnapshot.saveToSharedPreferences(getApplication())
```

Comment:
Are we still saving the selection of payment sheet/flowcontroller/customersheet?

Saving as customer sheet might lead to some unexpected behavior.

Reply:
Tested it in the UI PR. Added some lines in `PlaygroundSettings` to reset to a known value if that value is not available for `CustomerSheet` or vice versa. The `defaultValue` isn't based on the configuration data. I'll make a task to do that after adding end-to-end tests.

Reply:
The customer ID being available across configurations won't work between `PaymentSheet` and `CustomerSheet` and `FlowController` and `CustomerSheet` but will between `PaymentSheet` and `FlowController`. I'll fix this in another PR as well.